src/test/mod.rs

```diff
@@ -27,17 +27,25 @@ use self::harness::{expect_messages, ExpectedMessage, init_env, mock_server, moc
 
 use url::Url;
 use ls_types::*;
+use lsp_data::InitializationOptions;
 use std::path::Path;
 
 impl ServerMessage {
     pub fn initialize(id: usize, root_path: Option<String>) -> ServerMessage {
+         Self::initialize_with_opts(id, root_path, None)
+    }
+
+    pub fn initialize_with_opts(id: usize, root_path: Option<String>, initialization_options: Option<InitializationOptions>) -> ServerMessage {
```

Comment:
@nrc not sure I understand. This creates only a client-specific message. Since server itself has no use for it, it's used only in tests and once for our CLI "client". [This](https://github.com/rust-lang-nursery/rls/pull/420/files/d0de844d01a79cd93a9ec007aea3129e0c900859#diff-4ce93534efc34e923ce01e975eb7ed80R400) is where we parse the additional options when the server receives the `initialize` request via JSON-RPC and [here](https://github.com/rust-lang-nursery/rls/pull/420/files/d0de844d01a79cd93a9ec007aea3129e0c900859#diff-53ac0aa2d451315631bac6f85bcfb7bdR68) we're passing it to `init` action handler.